crates/sui-types/src/intent.rs

```diff
@@ -0,0 +1,238 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::batch::AuthorityBatch;
+use crate::messages::{TransactionData, TransactionEffects};
+use crate::messages_checkpoint::CheckpointSummary;
+
+use serde::{Deserialize, Serialize};
+
+/// Current Sui version (for compatibility purposes).
+/// TODO: Increment value when a non-backwards compatible Sui version is launched (i.e. when
+/// changing the format, signature or field ordering of signable types).
+const SUI_COMPATIBILITY_VERSION: u8 = 0;
+const SUI_CHAIN_ID: u8 = 0;
```

Comment:
These shouldn't be constants but instead we should use enums with `repr(u8)` in order to better control their values via the type system (invalid values can't even be represented)

Reply:
The plan as explained is to get these from some config or epoch change etc, this is a temp solution.